docs/modules/community/pages/release.adoc

```diff
@@ -15,3 +15,137 @@ Don't forget to add your key to https://downloads.apache.org/james/KEYS
 
     $ ssh people.apache.org
     $ cd /www/www.apache.org/dist/james
+
+== Release cheat sheet
+
+First, discuss your intention to perform a release on server-dev mailing list, to ensure a consensus would be reached
+before starting this time-consuming process.
+
+Then you can prepare the release materials:
+
+ * Ensure the `CHANGELOG.md` and `upgrade-instructions.md` files are up to date.
+ * Freeze your version in it and initiate a new iteration.
+ * Write a blog post within `/src/homepage/_post` about your release.
+ * Update the documentation website `/docs`, `src/site`, and `README.md` to point to your new release.
+ * Update the `/dockerfiles`
+
+Open a pull request to validate the changes.
+
+Once merged, it is time for the maven release.
+
+Set up your maven configuration in `~/.m2/settings.xml` :
+
+[source,bash]
+----
+<settings>
+  <servers>
+    <!-- To publish a snapshot of some part of Maven -->
+    <server>
+      <id>apache.snapshots.https</id>
+      <username>apacheId</username>
+      <password>xxxxxx</password>
+    </server>
+    <!-- To stage a release of some part of Maven -->
+    <server>
+      <id>apache.releases.https</id>
+      <username>apacheId</username>
+      <password>xxxxxxxx</password>
+    </server>
+    <server>
+      <id>svn.apache.org</id>
+      <username>apacheId</username>
+      <password>xxxxxx</password>
+    </server>
+
+  </servers>
+</settings>
+----
+
+Run these commands to perform a release, from the master branch:
+
+[source,bash]
+----
+mvn clean install -DskipTests -Djib.skip -Dmaven.javadoc.skip=true -T 4
```

Comment:
Ensure that HEAD has a clean build on CI ? 

I realize that mvn release creates commits of its own to mess around with the version numbers in the pom files but I feel it would be cleaner to have a green build on ci as a basis since we skip tests in the release

Reply:
Agree to explicitly add this.